packages/textfield/src/mwc-textfield.scss

```diff
@@ -28,6 +28,10 @@ limitations under the License.
   @extend %material-icons;
 }
 
+.hidden, .hidden::before, .hidden::after {
+  display: none;
```

Comment:
Would `visibility:hidden` be better? Would it actually avoid this whole issue, because `visibility:hidden` shouldn't affect the computed size?

Reply:
In this case, the display has to be none or else if you don't set a helper or a char counter, you'll have excess whitespace under the textfield. Also the problem is when they set `<mwc-textfield>` to display none, not what we do internally.

Reply:
Though, we do need the elements to actually exist in the DOM or else we have to deal with recreating the char counter foundation or the foundation needing the char counter div when it is not rendered, and working around this would complicate the js logic more than this styling.

Reply:
> In this case, the display has to be none or else if you don't set a helper or a char counter, you'll have excess whitespace under the textfield.

Not sure I understand, why would `visibility:hidden` cause extra whitespace? Doesn't `hidden` mean layout is normal but the browser just doesn't draw anything?

> Also the problem is when they set `<mwc-textfield>` to display none, not what we do internally.

Yeah, makes sense, though if our `.hidden` rule could use `display:hidden`, then we could recommend users use the `hidden` attribute instead of setting `display:none` in order to avoid this problem.

Reply:
For the first part: the styles in `MDCTextfield` add specific height styling to the `.mdc-text-field-helper-line` if it is simply present. If we set `visibility:hidden`, then the underbar still has that height since `visibility` does not change the height unlike `display:none`.

For the second one, `visibility:hidden` does not fit all the use cases of `display: none`. What if they don't want it to be visible and for it to not take up space on the page.